### PR TITLE
Feat/zapper v3

### PR DESCRIPTION
**Pull request by vuonghuuhung**
## Release Notes

- **New Features**
  - Introduced a new dependency, `@oraichain/oraidex-common`, enhancing functionality.
  - Updated the `indexer_v3` URL to transition to a staging environment.

- **Improvements**
  - Enhanced token price handling in the Create Position Form for improved accuracy in calculations.
  - Updated error handling in the Zap Out Form for clearer feedback during operations.
  - Increased debounce time for zap amount input to improve performance.
  - Refined logic in the `useGetPoolDetail` hook for better pool detail retrieval.

- **Bug Fixes**
  - Adjusted logic in the `simulateAprPosition` function to prevent errors when accessing undefined values.

- **Style**
  - Improved code organization and readability across multiple components.

- **Chores**
  - Removed unused imports in the Add Liquidity hook to reduce dependencies.

### CODE REVIEW

**Review by haunv3 — APPROVED** [review]

**Review by haunv3 — APPROVED** [review]

**Review by haunv3 — DISMISSED** [review]

File: src/pages/Pool-V3/components/PoolList/index.tsx [hunk 1]
```diff
@@ -83,16 +83,16 @@ const PoolList = ({ search }) => {
     (async () => {
       try {
         setLoading(true);
-
+        console.log({ theme, poolList, poolPrice });
         if (!poolList?.length) return;
```
Comment (trungbach) [inline src/pages/Pool-V3/components/PoolList/index.tsx:87]: Can you pls remove debug logs? @vuonghuuhung

File: src/pages/Pool-V3/components/PoolList/index.tsx [hunk 2]
```diff
@@ -83,16 +83,16 @@ const PoolList = ({ search }) => {
...
         if (poolList.length > 0 && isEqual(prevPoolListRef.current, poolList)) return;
-        const fmtPools = (poolList || []).map(formatPoolDataCallback).filter((e) => e.isValid);
-
+        const fmtPools = poolList.map(formatPoolDataCallback).filter((e) => e.isValid);
         setDataPool(fmtPools);
       } catch (error) {
         console.log('error: SingletonOraiswapV3 getPools', error);
       } finally {
         prevPoolListRef.current = poolList ?? [];
+        // console.log('prevPoolListRef.current', prevPoolListRef.current);
```
Comment (trungbach) [inline src/pages/Pool-V3/components/PoolList/index.tsx:95]: remove it also

File: src/pages/Pool-V3/components/PriceRangePlot/utils.ts [hunk 3]
```diff
@@ -116,6 +116,8 @@ export const getTickAtSqrtPriceFromBalance = (
   const minTick = getMinTick(spacing);
   const maxTick = getMaxTick(spacing);
 
+  const testPrice = calcPrice(isXtoY ? minTick : maxTick, isXtoY, xDecimal, yDecimal);
```
Comment (trungbach) [inline src/pages/Pool-V3/components/PriceRangePlot/utils.ts:119]: Where do you use this variable mate? @vuonghuuhung

File: src/pages/Pool-V3/components/ZapOutForm/index.tsx [hunk 4]
```diff
@@ -215,7 +227,7 @@ const ZapOutForm: FC<ZapOutFormProps> = ({
       client = await CosmWasmClient.connect(network.rpc);
       const zap = new ZapperQueryClient(client, ZAPPER_CONTRACT);
       zapFee = Number((await zap.protocolFee()).percent);
-    } catch (error) { }
+    } catch (error) {}
```
Comment (trungbach) [inline src/pages/Pool-V3/components/ZapOutForm/index.tsx:230]: i think in this catch block we need at least console error for debug?

**Review by trungbach — CHANGES_REQUESTED** [review]

**Review by haunv3 — APPROVED** [review]

**Review by trungbach — APPROVED** [review]
LGTM. Good job mate.

**Review by haunv3 — APPROVED** [review]

**Review by quangdz1704 — APPROVED** [review]

**Pull request merged**